pkg/kubelet/kubelet.go

```diff
@@ -1458,12 +1467,18 @@ func (kl *Kubelet) GetKubeletContainerLogs(podFullName, containerName, tail stri
 
 // GetBoundPods returns all pods bound to the kubelet and their spec
 func (kl *Kubelet) GetBoundPods() ([]api.BoundPod, error) {
-	return kl.pods, nil
+	kl.podLock.RLock()
+	defer kl.podLock.RUnlock()
+	var result []api.BoundPod
+	copy(kl.pods, result)
```

Comment:
If this copy is really necessary, then be warned that you're doing a shallow copy here and any remaining bugs are going to be super hard to figure out. I think we should be OK if we treat this as immutable, though.